kafka-connector/src/main/java/com/google/pubsub/kafka/sink/CloudPubSubGRPCPublisher.java

```diff
@@ -56,7 +56,8 @@ private void makePublisher() {
       log.info("Creating publisher.");
       publisher = PublisherGrpc.newFutureStub(ConnectorUtils.getChannel());
       // We change the publisher every 25 - 35 minutes in order to avoid GOAWAY errors.
-      nextPublisherResetTime = System.currentTimeMillis() + rand.nextInt(10 * 60 * 1000) + 25 * 60 * 1000;
+      nextPublisherResetTime =
```

Comment:
Is there no way to check on the channel itself, like by calling  isTerminated?

Reply:
It's a possibility we might be able to do that in the future. A second change will have to be made to not throw exceptions in the CloudPubSubSinkTask when flushing messages. When an exception is thrown, everything shuts down. That means introducing retry and making some significant changes to the class. Once we have that, then we can throw an exception when we have an error, detect when termination occurs, and recreate the publisher.

There is some urgency for a fix to this problem for a customer and so my hope was to use this to solve the problem in the meantime.